### PR TITLE
Add Bezier surface API

### PR DESCRIPTION
**Pull request by julianschuler**
This PR adds bezier surfaces to the high level API. While not using the same interface, this may be related to #47.

I also did a little deduplication and cleanup in the high level API, please let me know if I should factor this out to another PR.

### CODE REVIEW

**Comment by bschwind** [comment]
Thanks for the PR! And sorry for the code mess and duplication, that's on my mental todo list 😅

It would be great if you could add a little example of building something with this API in `crates/opencascade/examples` (I should probably add a top-level directory for the examples to live in) so I can see how it's used, and others can see what kind of shapes can be created with it.

**Comment by julianschuler** [comment]
No worries, I am really glad this project exists in the first place! Working with OpenCASCADE directly is really not fun and I am happy to contribute to a way more ergonomic API!

I now added an example where the bezier surface is used to create an airfoil like shape.

File: crates/opencascade/src/primitives.rs
Comment (bschwind) [inline crates/opencascade/src/primitives.rs:305]: We could make this API a bit more flexible:

```suggestion
    pub fn bezier<T: AsRef<[DVec3]>>(poles: impl IntoIterator<Item = T>) -> Self {
```

You'll then need to update various points in this function with calls to `.into_iter()` and `.as_ref()`

This should allow more flexibility in calling this function, allowing things like:

```rust
let surface = Surface::bezier([points1, points2]);
```

Comment (bschwind) [reply crates/opencascade/src/primitives.rs:305]: Or maybe we can have a doubly nested `impl Iterator`? I'll look into that.

Comment (bschwind) [reply crates/opencascade/src/primitives.rs:305]: @julianschuler ah, you can collect the iterator into a `Vec` for ease-of-use internally. It's more about presenting a nice API to end users and less about performance.

Comment (julianschuler) [reply crates/opencascade/src/primitives.rs]: Thanks for the clarification, I now implemented it that way.

**Comment by bschwind** [comment]
@julianschuler happy to merge this once the merge conflicts are resolved, sorry for the churn!

**Comment by julianschuler** [comment]
That was quite a rebase, but now everything should be fixed.

**Review by bschwind — APPROVED** [review]
LGTM, another great PR. I liked the `from_make_x` refactors, that's a good cleanup.

Thanks for the contribution!

**Pull request merged**
